### PR TITLE
Устинов - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by UAlex322**

### CODE REVIEW

**Comment by UAlex322** [comment]
@Dnoniel-Ermolaev @Faert примите участие в code review, пожалуйста.

**Review by Dnoniel-Ermolaev — APPROVED** [review]
Всё хорошо.

**Review by Faert — APPROVED** [review]
Всё хорошо

File: modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+#include <random>
+#include "include/complex_number.h"
+
+TEST(Ustinov_ComplexNumberTest, Equality1) {
+    ComplexNumber z1(3.0, 1.0),
+                  z2(3.0, 1.0);
+
+    bool eq = (z1 == z2);
+
+    EXPECT_EQ(eq, true);
```
Comment (dkurt) [inline modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp:13]: Should be
Expect_eq(z1, z2)

File: modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+TEST(Ustinov_ComplexNumberTest, Equality2) {
+    ComplexNumber z1(1.0, 1.0),
+                  z2(1.0, -1.0);
+
+    bool eq = (z1 == z2);
+
+    EXPECT_EQ(eq, false);
```
Comment (dkurt) [inline modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp:22]: Expect_ne

File: modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+TEST(Ustinov_ComplexNumberTest, Division3) {
+    ComplexNumber z1(1.0, 0.0);
+    ComplexNumber z2(1.0, 1.0);
+
+    ComplexNumber z1_div_by_z2 = z1 / z2;
+
+    EXPECT_EQ(z1_div_by_z2, ComplexNumber(0.5, -0.5));
+}
```
Comment (dkurt) [inline modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp:158]: Keep only one test per operation.

Comment (UAlex322) [reply modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp:158]: Should I test only one pair of numbers per operation or several pairs of numbers but in single test?

Comment (dkurt) [reply modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp:158]: I think one set of representative values is ok. Representative means values a different (real, imag parts and result).

File: modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+TEST(Ustinov_ComplexNumberTest, Equality) {
+    std::pair<ComplexNumber, ComplexNumber> eq_pair
+        {ComplexNumber(3.0, 1.0), ComplexNumber(3.0, 1.0)};
+    std::pair<ComplexNumber, ComplexNumber> neq_pair
+        {ComplexNumber(1.0, 1.0), ComplexNumber(1.0, -1.0)};
+
+    EXPECT_EQ(eq_pair.first, eq_pair.second);
+    EXPECT_NE(neq_pair.first, neq_pair.second);
+}
```
Comment (dkurt) [inline modules/complex-number/test/test_ustinov_alexandr_complex_number.cpp:17]: Too confusing approach, please refactor. Test only one thing at test.

**Pull request merged**